### ISSUE TITLE
Using selectAllFiltered() to select only the result of a filter model

### ISSUE THREAD

**Issue by jeffeld** [ceolter/ag-grid#1494]
Using ag-grid 6 we implemented our own "select all header". We have a bug where using it with a a filter model applied, resulted in all the rows being selected rather than just the result of the filter (which is the desired behaviour).

Checked back on the ag-grid documentation and discovered selectAllFiltered(). So I updated our project to ag-grid 8 and our "select all" code to this:

    if (this.isAllSelected)  {
        let filterModel = this.options.api.getFilterModel();
        if (ng.equals({}, filterModel)) {
            this.options.api.selectAll();
        } else {
            this.options.api.selectAllFiltered();
        }
    }

Basically it checks if there is a filter model, and if there is call selectAllFiltered(). However, selectAllFiltered() selects ***all*** the rows in the grid, and not just the result of the filter model.

Is my logic (and expectation here) correct? 

Thanks,
Jeff

**Comment by willhalling** [comment]
Hi @jeffeld 

We cannot reproduce this issue using our examples at https://www.ag-grid.com/javascript-grid-filtering/

Please can you provide a code example in [Plunker](https://plnkr.co/) to help us investigate further?

**Comment by jeffeld** [comment]
Hi @willhalling 

Thank you for the quick response; much appreciated!

False alarm! I found a logic bug elsewhere that was causing a selectAll() to be executed before the selectAllFiltered().

Thanks,
Jeff
